### DISCUSSION TITLE
Using ccache with '-specs=nano.specs'

### DISCUSSION THREAD

**Discussion by floitsch** [ccache/ccache#1385] [Questions & answers] [answered]
I'm trying to cross-compile with `arm-none-eabi-gcc`. Our project needs the flag `-specs=nano.specs`.

Unfortunately that seems to break ccache. In the logs I now have:
```
[2024-01-24T10:36:58.306578 191738] Failed to lstat nano.specs: No such file or directory
[2024-01-24T10:36:58.306581 191738] While processing -specs=nano.specs: nano.specs is missing
[2024-01-24T10:36:58.306602 191738] Failed; falling back to running the real compiler
```

I tried to add a `--ccache-skip` in front of `specs=nano.specs` but that doesn't seem to have any effect.

How can I tell ccache to ignore the nano.specs file?
Or is there a way to tell ccache where to find the file?

**Comment by jrosdahl** [comment, marked as answer]
It makes sense that `--ccache-skip` should have the effect you expected, but it currently doesn't. However, you can use [`ignore_options`/`CCACHE_IGNOREOPTIONS`](https://ccache.dev/manual/4.9.html#config_ignore_options) instead.

Another workaround could be to specify the full path to nano.specs, e.g. with `-specs=/usr/lib/arm-none-eabi/lib/nano.specs`.
